src/sql/workbench/parts/jobManagement/electron-browser/jobsView.component.ts

```diff
@@ -945,6 +947,11 @@ export class JobsViewComponent extends JobManagementView implements OnInit, OnDe
 
 	public async openCreateJobDialog() {
 		let ownerUri: string = this._commonService.connectionManagementService.connectionInfo.ownerUri;
-		await this._commandService.executeCommand('agent.openJobDialog', ownerUri);
+		if (!this._isDialogOpen) {
```

Comment:
I wonder if it's a cleaner pattern to have the `agent.openJobDialog` command (and related open dialog commands) maintain the `isDialogOpen` state and simply ignore the request if the dialog is already open.  It seems like the state is getting spread around with this approach, rather than being more coupled with the most closely related functionality.  i.e. is there a scenario where we'd want `agent.openJobDialog`  to open another dialog when the dialog is already being displayed, if not, the state probably belongs inside the command handler.

Reply:
That's a good idea. I think I might add this feature to all the dialogs, since this seems like a common problem.